src/Appium.Net/Appium/Interfaces/IHasPerformanceData.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+
+namespace OpenQA.Selenium.Appium.Interfaces
+{
+    public interface IHasSupportedPerformanceData
+    {
+        /// <summary>
+        /// Returns the information of the system state which is supported to read as like cpu, memory, networ
+        /// </summary>
+        /// <param name="packageName">The package name of the application</param>
+        /// <param name="dataType">The type of system state which wants to read. It should be one of the supported performance data types.</param>
+        /// <param name="dataReadTimeout">The number of attempts to read (optional)</param>
+        /// <returns>
+        /// Table of the performance data, The first line of the table represents the type of data.
+        ///         The remaining lines represent the values of the data.
+        ///         in case of battery info : [[power], [23]]
+        ///        in case of memory info :
+        ///             [[totalPrivateDirty, nativePrivateDirty, dalvikPrivateDirty, eglPrivateDirty, glPrivateDirty,
+        ///                       totalPss, nativePss, dalvikPss, eglPss, glPss, nativeHeapAllocatedSize, nativeHeapSize],
+        ///                      [18360, 8296, 6132, null, null, 42588, 8406, 7024, null, null, 26519, 10344]]
+        ///        in case of network info :
+        ///            [[bucketStart, activeTime, rxBytes, rxPackets, txBytes, txPackets, operations, bucketDuration,],
+        ///                      [1478091600000, null, 1099075, 610947, 928, 114362, 769, 0, 3600000],
+        ///                      [1478095200000, null, 1306300, 405997, 509, 46359, 370, 0, 3600000]]
+        ///        in case of network info :
+        ///            [[st, activeTime, rb, rp, tb, tp, op, bucketDuration],
+        ///                       [1478088000, null, null, 32115296, 34291, 2956805, 25705, 0, 3600],
+        ///                      [1478091600, null, null, 2714683, 11821, 1420564, 12650, 0, 3600],
+        ///                      [1478095200, null, null, 10079213, 19962, 2487705, 20015, 0, 3600],
+        ///                      [1478098800, null, null, 4444433, 10227, 1430356, 10493, 0, 3600]]
+        ///        in case of cpu info : [[user, kernel], [0.9, 1.3]]
+        /// </returns>
+        IList<object> GetSupportPerformanceData(string packageName, string dataType, int dataReadTimeout = 0);
```

Comment:
does zero value mean the timeout is not limited?

Reply:
I can't even remember the specification for the timeout. When I do, i'll add it to the xml documentation

Reply:
actually, I've jus checked the source code. The argument name is confusing. The value means how many times the perf data retrieval operation should be retried in case of failure: https://github.com/appium/appium-android-driver/blob/c1a8ee54b69e14940d4c2ecb766be7f75235aba2/lib/commands/performance.js#L69

The default value is 2 with 1000ms delays between each retry

Reply:
If I set this to 0 the appium logs doesn't display a second attempt, it just returns null.
1. I'm going to change the parameter to dataReadAttempts and update the param notes to make this clearer. 
2. Set the default value of dataReadAttempts to 1 because setting it to 0 returns null from the server (1.18).
3. Prevent users from setting 0 as the default value

Reply:
made the above changes

Reply:
I would rather create an overloaded method without this parameter. As far as I can see it is optional, so you could just skip sending it to the server: https://github.com/appium/appium-base-driver/blob/eba4635dcd8fe81899c3349d7e474acfb66037e7/lib/protocol/routes.js#L380

Reply:
I could either create an overload or perform a check on the driver method code, as it's optional, it already works as an overload to users

Reply:
is it? We are always sending the `dataReadTimeout` arg in https://github.com/appium/appium-dotnet-driver/pull/422/files#diff-db1821172278290200c0e11b4a1a7b1dR120

Reply:
changes made

Reply:
An alternative is making the dataReadAttempts argument nullable on the driver like so:

` public IList<object> GetPerformanceData(string packageName, string performanceDataType,
            int? dataReadAttempts = null) `
and checking for the value in the android command executor like so:

`public static object[] GetPerformanceData(IExecuteMethod executeMethod, string packageName, string dataType, int? dataReadTimeout)`

      `{
            var requiredKeys = new [] {"packageName", "dataType"};
            var requiredKeyValues = new object[] {packageName, dataType};
            if (dataReadTimeout is null)
                return executeMethod.Execute(AppiumDriverCommand.GetPerformanceData,
                    PrepareArguments(requiredKeys, requiredKeyValues)).Value as object[];
            requiredKeys.SetValue("dataReadTimeout", 2);
            requiredKeyValues.SetValue(dataReadTimeout, 2);
            return executeMethod.Execute(AppiumDriverCommand.GetPerformanceData,
                PrepareArguments(requiredKeys, requiredKeyValues)).Value as object[];
        }`